src/table-view/table-view.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import createReactClass from 'create-react-class'
+import PropTypes from 'prop-types'
+
+import { default as Table } from '../table'
+import { default as Select } from '../select'
+import { default as Button } from '../button'
+
+export default createReactClass({
+  displayName: 'TableView',
+  propTypes: {
+    rows: PropTypes.array,
+    columns: PropTypes.array,
+    filters: PropTypes.array,
+    onFilterChange: PropTypes.func,
+    onClearFilters: PropTypes.func
+  },
+  getDefaultProps() {
+    return {
+      rows: [],
+      columns: [],
+      filters: []
+    }
+  },
+
+  render() {
+    const { rows, columns, onFilterChange, onClearFilters } = this.props
+    const filters = sanitizeFilters(this.props.filters)
+
+    return (
+      <div style={{ display: 'flex', flexDirection: 'column' }}>
+        <div style={{ display: 'flex', width: '50%', alignItems: 'stretch' }}>
+          {filters.map(f => (
+            <div style={{ flex: '1', marginLeft: '10px' }}>
+              <Select
+                key={f.dataKey}
+                multiple
+                label={f.label}
+                value={f.value}
+                items={extractFilterValues(rows, f)}
+                onChange={e => onFilterChange(f.dataKey, e.target.value)}
+              />
+            </div>
+          ))}
+
+          <Button style={{
+            flex: 1,
+            marginLeft: '10px',
+            visibility: allFiltersEmpty(filters) ? 'hidden' : 'visible'
+          }} fullWidth variant='contained' onClick={onClearFilters}>
+            Clear Filters
+          </Button>
+        </div>
+        <Table rows={filterRows(rows, filters)} columns={columns} />
```

Comment:
Same as above, even if the items of the table didn't change it will still call `filterRows` on every re-render, this and the above causes performance implications when dealing with bigger data tables

Reply:
Similar elaboration as [above](https://github.com/jayesh-bhoot/challenge-table-view/pull/1#discussion_r309992180)